lib/AST/RequirementMachine/PropertyMap.cpp

```diff
@@ -140,11 +140,11 @@ void PropertyBag::dump(llvm::raw_ostream &out) const {
 /// \p lookupTerm exactly equals the key.
 MutableTerm
 PropertyBag::getPrefixAfterStrippingKey(const MutableTerm &lookupTerm) const {
-  assert(lookupTerm.size() >= Key.size());
+  ASSERT(lookupTerm.size() >= Key.size());
   auto prefixBegin = lookupTerm.begin();
   auto prefixEnd = lookupTerm.end() - Key.size();
-  assert(std::equal(prefixEnd, lookupTerm.end(), Key.begin()) &&
-         "This is not the bag you're looking for");
+  DEBUG_ASSERT(std::equal(prefixEnd, lookupTerm.end(), Key.begin()) &&
```

Comment:
Why `DEBUG_ASSERT` here?   (Genuinely curious.)

Reply:
It seemed that in a few places the asserts are extremely defensive because the caller establishes the condition immediately... but maybe that line of reasoning defeats the purpose of asserts!

I also used DEBUG_ASSERT in a few 'inner loops'. I was going to profile this patch anyway so I can play around with things a bit to see where we stand.

Reply:
Thanks for that clarification.   It helps to hear people talk about how they think of these things.